### DISCUSSION TITLE
Anchor links are not generated

### DISCUSSION THREAD

**Discussion by axilleas** [allejo/jekyll-toc#38] [Q&A] [answered]
**Public Repo**

https://github.com/axilleas/home-assistant.github.io/pull/2

**Description**

Hi! I'm trying to contribute to the home assistant repo and add table of contents to the integrations pages, for example https://www.home-assistant.io/integrations/tradfri/. While the table of contents is generated, the anchor links are not. Am I missing something?

**Your Usage**

Markdown file example https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/tradfri.markdown

```twig
{% include toc.html html=content sanitize=true h_min=2 %}
```

**Comment by allejo** [comment, marked as answer]
Take a look at the rendered page and inspect the headings,

```html
<h3>
  <a class="title-link" name="configuration-variables" href="#configuration-variables"></a> Configuration Variables
</h3>
```

The IDs are being created as `<a>` tags inside of the `<h3>`. The way this snippet works is that it [relies on the `<h1>` - `<h6>` tags having the `id` attribute](https://allejo.io/blog/a-jekyll-toc-without-plugins-or-javascript/#how-it-works).

You would have to modify this project to instead look for `<a>` tags and then pull out the `name` attribute values.

**Comment by axilleas** [comment]
Ah, I see. I figured it had something to do with using CommonMark instead of Kramdown. Thanks :)
